### ISSUE TITLE
Group by functionality in table

### ISSUE THREAD

**Issue by brakmic-aleksandar** [ArthurSonzogni/FTXUI#327]
Hey,
I'd like to implement group by functionality for Table. Where you could do something like this:
```cpp
vector<vector<string> data { 
  {"Id", "Name", "Surname"}, 
  {"0", "Aleksandar", "Brakmic"}, 
  {"1", "Arthur", "Sonzogni"}, 
  {"2", "Bosiljka", "Brakmic" } 
};
auto tbl = Table(data) | group_by("Surname");
```

Rendered output would look something like this

```

Id       Name

Surname: Brakmic
-----------------------------
0        Aleksandar
2        Bosiljka

Surname: Sonzogni
----------------------------
1        Arthur
```
It would consider first data element a header row.
Order by  functionality like this would be nice aswell. :)

Is that something that would fit into this library or is it out of scope of it?

**Comment by ArthurSonzogni** [comment]
Interesting idea!

1. If we allow a table to be grouped by, then there are multiple sub-table. Then the API to select rows/columns/rectangle and decorate the table seems not working anymore. Maybe this should be build outside of Table?
2. Table rely on `gridbox` for positioning elements, and separator/borders correctly. Here, I guess you want all the sub-Table to share column width. At the same time, the group are displayed outside of the gridbox. This seems a bit complex to reuse.

If the API is nice and the code looks nice to me, then I wouldn't be against including it into FTXUI. However, there is a challenge here. I am not sure this would be possible to hook into the existing `table`.

**Comment by brakmic-aleksandar** [comment]
Yeah, exactly group_by would return new table with only one column which would contain tables as data, decorating subtables could be done throught callbacks (although i'm not sure whether i like that idea tbh), at least thats the idea for now.
Also another feature i've had in mind was that we could add actual column header definitions to table, it would make implementing scrolling functionality for table much easier, at the moment there is no way (AFAIK) to freeze column headers/first row.

**Comment by brakmic-aleksandar** [comment]
Other thing that could be done is to have GroupedTable and order_by would transfrom Table into it, but i don't like that idea that much, it would be clean on the outside, but i don't like having multiple table types.
`auto tbl = Table(data) | order_by("Surname")`

**Comment by ArthurSonzogni** [comment]
`Table` is part of ftxui::dom. This is to print a single frame.
It seems you might be interested in building an interactive component, and be part of `ftxui::component` instead.

I see several ideas:
- Navigating inside the table.
- Freezing the first column.
- Group by.
- Sort.
- Filtering.

It is going to be a bit hard to fit everyone wishes, and style requirements. Still, it would be an awesome component.
Maybe I should add a "extension" section to FTXUI readme and reference every components that are aren't really building blocks, but more opinionated/complex ones.

**Comment by brakmic-aleksandar** [comment]
Yeah, extension test is nice idea, it could be also used like "experimental" in C++ stl where in future releases you could move really used/liked components to base namespace.
Also, yeah, this kind of functionality will definetly require new component, it could be named InteractiveTable, that name should make clear distinction between the two.
